spec/factories/course_condition_achievements.rb

```diff
@@ -3,5 +3,6 @@
           class: Course::Condition::Achievement.name, aliases: [:achievement_condition] do
     course
     achievement
+    association :conditional, factory: :course_achievement
```

Comment:
@dariusf I think you can leave it as referencing an achievement. Just make sure the generated achievement belongs to the same course.